backend/infrahub/core/query/diff.py

```diff
@@ -668,7 +668,7 @@ async def query_init(self, db: InfrahubDatabase, **kwargs: Any) -> None:
             AND (from_time <= diff_rel.from < $to_time)
             AND (diff_rel.to IS NULL OR (from_time <= diff_rel.to < $to_time))
             AND r_root.from <= diff_rel.from
-            AND (r_root.to IS NULL OR r_root.to >= diff_rel.from)
+            AND (r_root.to IS NULL OR diff_rel.branch <> r_root.branch OR r_root.to >= diff_rel.from)
```

Comment:
this little change is required to account for this case
1. make branch `branch`
2. delete `NodeA` on `main`
3. update a relationship to `NodeA` on `branch`, where `NodeA` is not deleted